libtenzir/builtins/formats/kv.cpp

```diff
@@ -23,6 +23,37 @@ namespace {
 
 constexpr auto docs = "https://docs.tenzir.com/formats/kv";
 
+/// @brief Checks whether the index `idx` in `text` is escaped
+/// TODO the precise-parsers PR contains this in string.hpp
+inline auto is_escaped(size_t idx, std::string_view text) -> bool {
+  if (idx >= text.size()) {
+    return false;
+  }
+  // An odd number of preceding backslashes means it is escaped, for example:
+  // `x\n` => true, `x\\n` => false, `x\\\n` => true, 'x\\\\n' => false.
+  auto backslashes = size_t{0};
+  while (idx > 0 and text[idx - 1] == '\\') {
+    ++backslashes;
+    --idx;
+  }
+  return backslashes % 2 == 1;
+}
+
+[[nodiscard]] auto trim_quotes(std::string_view txt) -> std::string_view {
```

Comment:
It's a bit odd to me. that this function does not unescape. For example:

```
key="\"\\"
```


This currently parses as `\"\\`, but I'd have expected the value to be `"\`.